micromamba/tests/conftest.py

```diff
@@ -72,6 +67,10 @@ def keep_in_path(
     path_list = [p for p in path_list if keep_in_path(p)]
     os.environ["PATH"] = os.pathsep.join(path_list)
 
+    saved_environ = os.environ.copy()
```

Comment:
The previous impl did not restore `PATH`. I assumed that to be a mistake and fixed the code accordingly.